scripts/solutions.sh

```diff
@@ -0,0 +1,492 @@
+    shift
+done
+
+TMPFILES=$(mktemp -d)
+
+mkdir -p "$(dirname "$LOGFILE")"
+
+exec > >(tee -ia "$LOGFILE") 2>&1
+
+cleanup() {
+    rm -rf "$TMPFILES" || true
+}
+
+trap cleanup EXIT
+
+BASE_DIR=$(dirname "$(readlink -f "${BASH_SOURCE[0]}")")
+# shellcheck disable=SC1090
+. "$BASE_DIR"/common.sh
+
+check_command_mandatory_options() {
+    local -a missing_options=()
+    local -r command=$1
+
+    for option in ${COMMAND_MANDATORY_OPTIONS[$command]:-}; do
+        [[ ${!OPTIONS_MAPPING[$option]:-} ]] || missing_options+=("$option")
+    done
+
+    if (( ${#missing_options[@]} )); then
+        echo 1>&2 "Missing options for command '$command':" \
+                  "${missing_options[@]}"
+        return 1
+    fi
+
+    return 0
+}
+
+salt_minion_exec() {
+    salt-call "$@" --retcode-passthrough
+}
+
+salt_master_exec() {
+    SALTENV=${SALTENV:-$(get_salt_env)}
+    SALT_MASTER_CONTAINER_ID=${SALT_MASTER_CONTAINER_ID:-$(get_salt_container)}
+    crictl exec -i "$SALT_MASTER_CONTAINER_ID" "$@" saltenv="$SALTENV"
+}
+
+activate_solution() {
+    run "Updating Solutions configuration file" \
+        salt_minion_exec metalk8s_solutions.activate_solution \
+        solution="$NAME" \
+        version="$VERSION" \
+        --local
+
+    run "Deploying Solution components" \
+        salt_master_exec salt-run state.orchestrate \
+        metalk8s.orchestrate.solutions.deploy-components \
+        pillar="{'bootstrap_id': '$(get_salt_minion_id)'}"
+}
+
+deactivate_solution() {
+    run "Updating Solutions configuration file" \
+        salt_minion_exec metalk8s_solutions.deactivate_solution \
+        solution="$NAME" \
+        --local
+
+    run "Removing Solution components" \
+        salt_master_exec salt-run state.orchestrate \
+        metalk8s.orchestrate.solutions.deploy-components \
+        pillar="{'bootstrap_id': '$(get_salt_minion_id)'}"
+}
+
+configure_archives() {
+    local removed=${1:-False}
+
+    for archive in "${ARCHIVES[@]}"; do
+        salt_minion_exec metalk8s_solutions.configure_archive \
+            archive="$archive" \
+            removed="$removed" \
+            --local
+    done
+
+    salt_minion_exec saltutil.refresh_pillar
+}
+
+import_solution() {
+    SALTENV=${SALTENV:-$(get_salt_env)}
+
+    run "Updating Solutions configuration file" configure_archives
+    run "Importing Solutions" \
+        salt_minion_exec state.sls metalk8s.solutions.available \
+        saltenv="$SALTENV"
+}
+
+unimport_solution() {
+    SALTENV=${SALTENV:-$(get_salt_env)}
+
+    run "Updating Solutions configuration file" configure_archives True
+    run "Unimporting Solutions" \
+        salt_minion_exec state.sls metalk8s.solutions.available \
+        saltenv="$SALTENV"
+}
+
+namespace_is_in_environment() {
+    local -r namespace=$1 environment=$2
+
+    kubectl get namespace --selector="$ENV_LABEL=$environment" \
+        --output=jsonpath='{ .items[*].metadata.name }' \
+        | grep --quiet --word-regexp "$namespace"
```

Comment:
I think you should add a `KUBECONFIG` default value at the beginning of the script because here that will fail if you do not have any kubeconfig exported

Reply:
True, didn't thought about it since I was testing on an environment with an exported KUBECONFIG env var.

Reply:
done